sql_grader/models.py

```diff
@@ -23,22 +18,26 @@ class SqlProblem:
     answer_result = None
     is_ordered = True
 
+    # pylint: disable=R0913
     def __init__(
             self,
-            database,
-            answer_query,
+            database=None,
+            answer_query=None,
+            dataset=None,
             verify_query=None,
             is_ordered=True,
     ):
         """
         Initialize variables
         """
         self.database = database
+        if dataset:
+            self.database = create_database(dataset)
```

Comment:
Maybe we could move all this out of the `__init__` and run the the `create_database` call (under codejail) in the views/handlers, along with the other student code being executed? Capa codejailing doesn't trust course authors to respect the limits either.

Reply:
The `__init__` is called within codejail, so `create_database` and the instructor's query are executed safely. 
This also means that we won't have a couple of open sqlite databases in memory at all times (previously, they were cached in a module level dict).